daemon/src/collab_settlement/maker.rs

```diff
@@ -0,0 +1,264 @@
+    }
+
+    async fn handle(&mut self, msg: Accept) -> Result<()> {
+        let Accept { order_id } = msg;
+
+        let (mut framed, transaction, proposal, _peer) =
+            self.pending_protocols
+                .remove(&order_id)
+                .with_context(|| format!("No active protocol for order {order_id}"))?;
+
+        let mut tasks = Tasks::default();
+        tasks.add_fallible(
+            {
+                let executor = self.executor.clone();
+                async move {
+                    executor
+                        .execute(order_id, |cfd| {
+                            cfd.accept_collaborative_settlement_proposal(&proposal)
+                        })
+                        .await?;
+
+                    framed
+                        .send(ListenerMessage::Msg1(Decision::Accept))
+                        .await
+                        .context("Failed to send Msg1::Accept")?;
+
+                    // TODO: We will need to apply a timeout to these. Perhaps we can put a timeout
+                    // generally into "reading from the substream"?
+                    let Msg2 { dialer_signature } = framed
+                        .next()
+                        .await
+                        .context("End of stream while receiving Msg2")?
+                        .context("Failed to decode Msg2")?
+                        .into_msg2()?;
+
+                    let listener_signature = transaction.own_signature();
+
+                    let settlement = transaction
+                        .recv_counterparty_signature(dialer_signature)
+                        .context("Failed to receive counterparty signature")?
+                        .finalize()
+                        .context("Failed to finalize transaction")?;
+
+                    tracing::trace!(
+                        ?settlement,
+                        "Received collab settlement transaction from taker"
+                    );
+
+                    framed
+                        .send(ListenerMessage::Msg3(Msg3 { listener_signature }))
+                        .await
+                        .map_err(|source| Failed::AfterReceiving {
+                            source: anyhow!(source),
+                            settlement: settlement.clone(),
+                        })?;
+
+                    emit_completed(order_id, settlement, &executor).await;
+                    Ok(())
+                }
+            },
+            {
+                let executor = self.executor.clone();
+                move |failed| async move {
+                    match failed {
+                        Failed::BeforeReceiving { source } => {
+                            emit_failed(order_id, source, &executor).await;
+                        }
+                        Failed::AfterReceiving { source, settlement } => {
+                            // TODO: proceed with the transaction when taker will be able to handle
```

Comment:
❓I don't understand this TODO, could you elaborate what you mean here? Is this still relevant?

Reply:
it's still relevant and tracked by a separate ticket: #2157